backend/src/apiserver/server/api_converter_test.go

```diff
@@ -2306,3 +2314,2356 @@ func TestToApiRecurringRun(t *testing.T) {
+		},
+		{
+			UpdateTime: &timestamppb.Timestamp{Seconds: 100},
+			State:      apiv2beta1.RuntimeState_RUNTIME_STATE_UNSPECIFIED,
+			Error:      util.ToRpcStatus(util.NewInvalidInputError("Invalid input: %s", "sample value")),
+		},
+		{
+			UpdateTime: &timestamppb.Timestamp{Seconds: 100},
+			State:      apiv2beta1.RuntimeState_RUNTIME_STATE_UNSPECIFIED,
+			Error:      util.ToRpcStatus(util.NewInvalidInputError("Invalid input: %s", "sample value")),
+		},
+		{
+			UpdateTime: &timestamppb.Timestamp{Seconds: 100},
+			State:      apiv2beta1.RuntimeState_RUNTIME_STATE_UNSPECIFIED,
+			Error:      util.ToRpcStatus(util.NewInvalidInputError("Invalid input: %s", "sample value")),
+		},
+	}
+	got := toApiRuntimeStatuses(arg)
+	assert.Equal(t, expected, got)
+}
+
+func Test_toModelTask(t *testing.T) {
+	tests := []struct {
+		name    string
+		apiTask interface{}
+		want    *model.Task
+		wantErr bool
+		errMsg  string
+	}{
+		{
+			"V1 full spec",
+			&apiv1beta1.Task{
+				Id:              "1",
+				Namespace:       "ns1",
+				PipelineName:    "namespaces/ns1/pipelines/p1",
+				RunId:           "2",
+				MlmdExecutionID: "3",
+				CreatedAt:       &timestamppb.Timestamp{Seconds: 4},
+				FinishedAt:      &timestamppb.Timestamp{Seconds: 5},
+				Fingerprint:     "6",
+			},
+			&model.Task{
+				UUID:              "1",
+				Namespace:         "ns1",
+				PipelineName:      "namespaces/ns1/pipelines/p1",
+				RunId:             "2",
+				MLMDExecutionID:   "3",
+				CreatedTimestamp:  4,
+				StartedTimestamp:  4,
+				FinishedTimestamp: 5,
+				Fingerprint:       "6",
+				Name:              "",
+				ParentTaskId:      "",
+				State:             model.RuntimeStateUnspecified,
+				StateHistory:      nil,
+				MLMDInputs:        "",
+				MLMDOutputs:       "",
+				ChildrenPods:      nil,
+			},
+			false,
+			"",
+		},
+		{
+			"V2 full spec",
+			&apiv2beta1.PipelineTaskDetail{
+				RunId:       "2",
+				TaskId:      "1",
+				DisplayName: "task",
+				CreateTime:  &timestamppb.Timestamp{Seconds: 4},
+				StartTime:   &timestamppb.Timestamp{Seconds: 5},
+				EndTime:     &timestamppb.Timestamp{Seconds: 6},
+				// ExecutorDetail *PipelineTaskExecutorDetail `protobuf:"bytes,7,opt,name=executor_detail,json=executorDetail,proto3" json:"executor_detail,omitempty"`
+				State:       apiv2beta1.RuntimeState_CANCELING,
+				ExecutionId: 7,
+				Error:       util.ToRpcStatus(util.NewInvalidInputError("Sample error")),
+				Inputs: map[string]*apiv2beta1.ArtifactList{
+					"a1": {
+						ArtifactIds: []int64{1, 2, 3},
+					},
+				},
+				Outputs: map[string]*apiv2beta1.ArtifactList{
+					"b2": {
+						ArtifactIds: []int64{4, 5, 6},
+					},
+				},
+				ParentTaskId: "8",
+				StateHistory: []*apiv2beta1.RuntimeStatus{
+					{
+						UpdateTime: &timestamppb.Timestamp{Seconds: 9},
+						State:      apiv2beta1.RuntimeState_PAUSED,
+						Error:      util.ToRpcStatus(util.NewInvalidInputError("Sample error2")),
```

Comment:
What would be a real world example for `Sample error2`? and what's its relation with the `Sample error` above on line 3082? 
Given it's an `NewInvalidInputError`, do we expect the task to even have `StartTime` and `EndTime`?

Reply:
Good point. Removed the errors. Thank you.